src/Facebook/FileUpload/FacebookVideoUploader.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * Copyright 2014 Facebook, Inc.
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\FileUpload;
+
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Facebook;
+
+/**
+ * Class VideoUploadContext
+ * @package Facebook
+ */
+class VideoUploadContext
+{
+    /**
+     * @var string
+     */
+    public $accessToken;
+
+    /**
+     * @var string Video upload target Id, could be page_id || user_id || event_id || group_id
+     */
+    public $targetId;
+
+    /**
+     * @var string Upload file path
+     */
+    public $filePath;
+
+    /**
+     * @var int Upload file size in bytes.
+     */
+    public $fileSize;
+
+    /**
+     * @var int ID of the upload session
+     */
+    public $uploadSessionId;
+
+    /**
+     * @var int Start byte position of the next file chunk
+     */
+    public $startOffset;
+
+    /**
+     * @var int End byte position of the next file chunk
+     */
+    public $endOffset;
+
+    /**
+     * @var int ID of the video
+     */
+    public $videoId;
+}
+
+/**
+ * Class FacebookVideoUploader.
+ */
+class FacebookVideoUploader
+{
+    /**
+     * @var VideoUploadContext
+     */
+    protected $context;
+
+    /**
+     * @var int Max retry times
+     */
+    protected $maxTransferTries;
+
+    /**
+     * @var Facebook Reference to Facebook
+     */
+    protected $facebook;
+
+    public function __construct(Facebook $fb)
+    {
+        $this->facebook = $fb;
+    }
+
+    public function setFacebook($fb)
+    {
+        $this->facebook = $fb;
+    }
+
+    /**
+     * Upload video file by chunks
+     *
+     * @param string       $filePath
+     * @param int          $targetId
+     * @param string       $accessToken
+     * @param int          $maxTransferTries
+     * @return array
+     */
+    public function upload($filePath, $targetId, $accessToken, $maxTransferTries = 3)
```

Comment:
This class is an adapter which receive single file from use, cut it into chunks and feed facebook video upload api. We want to use this class simplify the developer from our chunked upload api(https://developers.facebook.com/docs/graph-api/video-uploads).

I'm not good at doc too. @SammyK , could you give some suggestion about how to improve the doc?


I would suggest to write the doc like this: 
+    /**
+     * Upload video file to Facebook, 
+     *
+     * @param string       $filePath
+     * @param int          $targetId
+     * @param string       $accessToken
+     * @param int          $maxTransferTries
+     * @return array
+     */